include/range/v3/detail/optional.hpp

```diff
@@ -86,6 +86,49 @@ namespace ranges
             }
         };
 
+        template<typename T>
+        struct optional<T &>
+        {
+        private:
+            variant<meta::nil_, T &> data_;
```

Comment:
No, because when instantiated with a reference type, `semiregular`'s two copy-assignment operators end up having the same signature --> hard error.